overrides/default.nix

```diff
@@ -888,8 +888,13 @@ lib.composeManyExtensions [
         }
       );
 
+      fastapi = prev.fastapi.overridePythonAttrs (old: {
+        # fastapi 0.111 depends on fastapi-cli, which depends on fastapi, resulting in infinite recursion
```

Comment:
Why are you using `final ? fastapi-cli`? You should always use `final.fastapi-cli`.

Reply:
`final.fastapi-cli` does not exist if `fastapi` version is 0.110.3

Reply:
In that case, don't branch on a dependency existing, branch on the version of `fastapi`.

Reply:
I think `final?fastapi-cli` is more robust than `prev.fastapi.version == "0.111.0"` because we don't know whether and when the upstream would fix the circular dependency.